src/native/public/mono/metadata/details/metadata-types.h

```diff
@@ -29,7 +29,7 @@ typedef enum {
 	MONO_CALL_THISCALL,
 	MONO_CALL_FASTCALL,
 	MONO_CALL_VARARG = 0x05,
-	/* unused, */
+	MONO_CALL_SWIFTCALL = 0x06,
```

Comment:
1. We should not extend this enum. it's supposed to mirror the 
metadata encoding.

2. This is the public Mono API, we shouldn't just add things here unless there's a need.

---

We should handle this the same way as "SuppressGCTransition": use a bit in `MonoMethodSignature` to note that it's a swift call -  see `MonoMethodSignature:suppress_gc_transition`.

Another idea is to make a `MonoExtCallConv` enum (not in this file, in `src/mono/metadata/metadata-internals.h`) and then do:

```c
typedef enum {
  MONO_EXT_CALLCONV_SUPPRESS_GC_TRANSITION = 0x01,
  MONO_EXT_CALLCONV_SWIFTCALL = 0x02,
  /// see MonoMethodSignature:ext_callconv - only 8 bits
} MonoExtCallConv;

....

typedef struct _MonoMethodSignature {
  ...
  uint8_t    ext_callconv; // see MonoExtCallConv 
} MonoMethodSignature;

static inline gboolean
mono_method_signature_has_ext_callconv (MonoMethodSignature *sig, MonoExtCallConv flags)
{
   return (sig->callconv_ext & flags) != 0;
}
```

Reply:
Thank you. Your proposal looks good to me. I implemented the `ext_callconv` as it offers extensibility for future flags.